cumulus_library/studies/core/core_templates/core_utils.jinja

```diff
@@ -1,38 +1,138 @@
+{#- A collection of macros for constructing common safe column selects
+
+    For layout, is is generally assumed that these are being invoked from within
+    a WITH TABLE AS(...) statement, so they will always assume two levels of
+    indentation.
+-#}
+
 {%- macro comma_delineate(loop) -%}
 {%- if not loop.last -%}
 , 
 {%- endif -%}
 {%- endmacro -%}
 
--- gets a list of columns from a known present source
+{#- gets a list of columns from a known present source -#}
 {% macro basic_cols(table, alias, targets) %}
         {%- for col in targets %}
         {{ alias }}.{{ col.lower() }}
         {{- comma_delineate(loop) }}
         {%- endfor %}
 {%- endmacro %}
 
+{#- gets a list of columns from a potentially null source.
+
+targets is an array expecting a data type of the following:
+    string: refers to a vanilla SQL primitive column
+    tuple(col, object, col_alias): refers to a one level nested SQL object
+    tuple(col, object, , child, col_alias): refers to a two level nested SQL object
 
--- gets a column, if present, from a basic SQL primitive col, or a null otherwise
-{% macro basic_nullable_cols(table, alias, targets, schema) %}
+-#}
+{% macro nullable_cols(table, alias, targets, schema) %}
         {%- for col in targets %}
-        {%- if schema[table][col] %}
+        {#- depth one nested column-#}
+        {%- if col is not string and col|length ==3%}
+        {%- if schema[table][col[0]][col[1]] %}
```

Comment:
nit: would it look too weird to indent the logic tree statements, but not the output stuff? Like:
```
    {%- if col is not string and col|length ==3%}
        {%- if schema[table][col[0]][col[1]] %}
        {{ alias }}.{{ col[0].lower() }}.{{ col[1].lower() }} AS {{col[2]}}
        {%- endif -%}
    {%- endif -%}
```

or
```
        {%- if col is not string and col|length ==3%}
            {%- if schema[table][col[0]][col[1]] %}
        {{ alias }}.{{ col[0].lower() }}.{{ col[1].lower() }} AS {{col[2]}}
            {%- endif -%}
        {%- endif -%}
```

I dunno - I wish jinja code was more readable in general.

Reply:
so - this is configurable as a jinja config param. i was explicitly asked to minimize the amount of indentation, but i agree that i would prefer to have if indentation. 

if we do, i want to do it as a seperate PR. this is already too big for my poor brain to reason over.